actions/src/subnet_actions.py

```diff
@@ -0,0 +1,56 @@
+from typing import Dict, Callable
+
+from openstack.exceptions import ResourceNotFound
+from openstack.network.v2.subnet import Subnet
+from st2common.runners.base_action import Action
+
+from openstack_api.openstack_network import OpenstackNetwork
+
+
+class SubnetActions(Action):
+    def __init__(self, *args, config: Dict, **kwargs):
+        """constructor class"""
+        super().__init__(*args, **kwargs)
+        self._api: OpenstackNetwork = config.get("openstack_api", OpenstackNetwork())
+
+    def run(self, submodule: str, **kwargs):
+        """
+        Dynamically dispatches to the method wanted
+        """
+        func: Callable = getattr(self, submodule)
+        return func(**kwargs)
+
+    def subnet_show(self, subnet):
+        """
+        Show subnet properties
+        :param subnet: Name or ID
+        :return: (status (Bool), reason/output (String/Dict))
+        """
+        try:
+            subnet = self.conn.network.find_subnet(subnet, ignore_missing=False)
```

Comment:
I don't see a conn variable being created here. How is it being accessed?

Reply:
This is the legacy way of using the API, I'll add a `raise NotImplementError` to make it clear :+1: